quickwit/quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -148,39 +148,43 @@ impl ConsumerContext for RdKafkaContext {
     fn pre_rebalance(&self, rebalance: &Rebalance) {
         if let Rebalance::Revoke(tpl) = rebalance {
             let partitions = collect_partitions(tpl, &self.topic);
-            debug!(partitions=?partitions, "Revoke partitions.");
+            debug!(partitions=?partitions, "revoke partitions");
 
             let (ack_tx, ack_rx) = oneshot::channel();
             return_if_err!(
                 self.events_tx
                     .blocking_send(KafkaEvent::RevokePartitions { ack_tx }),
-                "Failed to send revoke message to source."
+                "failed to send revoke message to source"
             );
-            return_if_err!(ack_rx.recv(), "Failed to receive revoke ack from source");
+            return_if_err!(ack_rx.recv(), "failed to receive revoke ack from source");
         }
         if let Rebalance::Assign(tpl) = rebalance {
             let partitions = collect_partitions(tpl, &self.topic);
-            debug!(partitions=?partitions, "Assign partitions.");
+            debug!(partitions=?partitions, "assign partitions");
 
             let (assignment_tx, assignment_rx) = oneshot::channel();
             return_if_err!(
                 self.events_tx.blocking_send(KafkaEvent::AssignPartitions {
                     partitions,
                     assignment_tx,
                 }),
-                "Failed to send assign message to source."
+                "failed to send assign message to source"
             );
             let assignment = return_if_err!(
                 assignment_rx.recv(),
-                "Failed to receive assignment from source."
+                "failed to receive assignment from source"
             );
-            for (partition, offset) in assignment {
-                let mut partition = tpl
-                    .find_partition(&self.topic, partition)
-                    .expect("Failed to find partition in assignment. This should never happen! Please, report on https://github.com/quickwit-oss/quickwit/issues.");
-                partition
-                    .set_offset(offset)
-                    .expect("Failed to convert `Offset` to `i64`. This should never happen! Please, report on https://github.com/quickwit-oss/quickwit/issues.");
+            for (partition_id, offset) in assignment {
+                let Some(mut partition) = tpl.find_partition(&self.topic, partition_id) else {
+                    warn!("partition `{partition_id}` not found in assignment");
+                    continue;
+                };
+                if let Err(error) = partition.set_offset(offset) {
```

Comment:
shouldn't we abort if there is an error on set_offset? (just a question, I don't know which kind of error we get from this call)